composer/models/huggingface.py

```diff
@@ -718,6 +836,59 @@ def write_huggingface_pretrained_from_composer_checkpoint(
     config = get_hf_config_from_composer_state_dict(composer_state_dict)
     config.save_pretrained(output_folder)
 
+    peft_config = get_peft_config_from_composer_state_dict(composer_state_dict)
+    if peft_config is not None:
+        peft_config.save_pretrained(str(output_folder))
+
     weights_state_dict = composer_state_dict['state']['model']
     torch.nn.modules.utils.consume_prefix_in_state_dict_if_present(weights_state_dict, prefix='model.')
-    torch.save(weights_state_dict, Path(output_folder) / 'pytorch_model.bin')
+
+    # NOTE: This only works for default adapter name
+    if peft_config is not None:
+        weights_state_dict = filter_state_dict_peft(weights_state_dict, peft_config)
+
+        torch.save(weights_state_dict, Path(output_folder) / 'adapter_model.bin')
+    else:
+        torch.save(weights_state_dict, Path(output_folder) / 'pytorch_model.bin')
+
+
+def filter_state_dict_peft(state_dict: Dict[str, Any],
+                           peft_config: 'PeftConfig',
+                           remove_adapter_names: bool = True) -> Dict[str, Any]:
+    """Filter a state dict to only include the weights needed for a PEFT model
+
+    Note: This function only works with LORA PEFT models right now.
+
+    Args:
+        state_dict (Dict[str, Any]): The state dict to filter
+        peft_config (PeftConfig): The PEFT config to use to filter the state dict
+        remove_adapter_names (bool, optional): Whether to remove the adapter names from the state dict keys. Defaults to True.
+
+    Returns:
+        Dict[str, Any]: The filtered state dict
+    """
+
+    if peft_config.peft_type != 'LORA':
+        raise NotImplementedError(f'Only LoRA PEFT is supported. Got {peft_config.peft_type}')
+
+    # Filtering copied from https://github.com/huggingface/peft/blob/4186c9b104644fd247a4cc0dc2dfc1ede4665204/src/peft/utils/save_and_load.py#L68C1-L86C116
```

Comment:
unfortunately that doesn't work because the function from the peft library takes in a `PeftModel`, which I don't have access to in all the places I need this function.